docs/rules/no-extraneous-import.md

```diff
@@ -4,7 +4,8 @@
 
 <!-- end auto-generated rule header -->
 
-If an `import` declaration's source is extraneous (it's not written in `package.json`), the program works in local, but will not work after dependencies are re-installed. It will cause troubles to your team/contributors.
+If an `import` declaration's source is extraneous (it's not written in `package.json`), the program works in local, but may not work after dependencies are re-installed. It will cause troubles to your team/contributors. If a declaration source is extraneous yet consistently works for you and your team, it may be a transitive dependency (a dependency of a dependency). Transitive dependencies should still be saved as an explicit dependency in `package.json` to avoid the risk of a dependency changing and removing the dependency of a dependency this `import` is relying on.
```

Comment:
**suggestion**: It would also would be worth adding a little bit about how a transitive dependency may be from a dev dependency

> Additionally, the transitive dependency could be a dev dependency, meaning your code could work in development but not in production!